src/main/java/io/jenkins/plugins/sample/OnboardingConfiguration.java

```diff
@@ -0,0 +1,181 @@
+        }
+    }
+
+    /**
+     * This method validates the name and username parameters based on validation regexes
+     * {@link #PLUGIN_NAME_REGEX_PATTERN} and {@link #USERNAME_REGEX_PATTERN}
+     * It just overrides the {@link jenkins.model.GlobalConfiguration#configure(StaplerRequest2, JSONObject)}} method
+     * to validate name (plugin name) and username fields when the Save or Apply Button is clicked on System Configuration page
+     * @param req
+     *      The stapler request object
+     * @param json
+     *      The JSON object that captures the configuration data for this {@link hudson.model.Descriptor}.
+     *      See <a href="https://www.jenkins.io/doc/developer/forms/structured-form-submission/">the developer documentation</a>.
+     * @return boolean If validation for the fields (name and username) is successfull, return true else return false
+     * @throws FormException
+     */
+    @Override
+    public boolean configure(StaplerRequest2 req, JSONObject json) throws FormException {
+        String pluginName = json.getString("name");
+        String userName = json.getString("username");
+
+        if (pluginName != null || userName != null) {
+            if (!PluginHelper.isMatch(pluginName, PLUGIN_NAME_REGEX_PATTERN)) {
+                return false;
+            }
+            if (!PluginHelper.isMatch(userName, USERNAME_REGEX_PATTERN)) {
+                return false;
+            }
+            req.bindJSON(this, json);
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * This method validates the url , username and password fields in the optional view on the Onboarding Plugin Section
+     * @param url
+     *      url to send username and password to
+     * @param username
+     *      username field
+     * @param password
+     *      password field
+     * @return
+     *      FormValidation object
+     * @throws IOException
+     * @throws ServletException
+     */
+    @POST
+    public FormValidation doTestConnection(
+            @QueryParameter("url") final String url,
+            @QueryParameter("username") final String username,
+            @QueryParameter("password") final String password)
+            throws IOException, ServletException {
+        try {
+            if (url.isEmpty() || username.isEmpty() || password.isEmpty()) {
+                return FormValidation.error("Mandatory parameters cannot be null or empty");
+            }
+            int statusCode = PluginHelper.httpPostBasicAuth(url, username, password);
+            if (statusCode != HttpURLConnection.HTTP_OK) {
+                return FormValidation.error("Server error : " + statusCode);
+            } else {
+                return FormValidation.ok("Input Validated");
+            }
+        } catch (IOException | InterruptedException e) {
+            return FormValidation.error("Server Error Occured with status code 500 ");
```

Comment:
I didn't run into this block when I had the server return HTTP 500 (instead I just got `Server error : 500`). Where is this claim coming from?

Reply:
That's different from HTTP 500. This error message is misleading.